geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -358,11 +371,51 @@ public LocatorClusterManagementService(ConfigurationPersistenceService persisten
     return list;
   }
 
+  @Override
+  public <A extends ClusterManagementOperation<V>, V extends JsonSerializable> ClusterManagementOperationResult<V> startOperation(
+      A op) {
+    ClusterManagementOperationResult<V> result = new ClusterManagementOperationResult<>();
+    result.setResult(executorManager.submit(op));
+    result.setUri(op.getUri());
+    result.setStatus(ClusterManagementResult.StatusCode.ACCEPTED, "async operation started");
+    return result;
+  }
+
+  /**
+   * this is intended for use by the rest controller. for Java usage, please use the Future returned
+   * by startOperation().getResult().
+   */
+  public <V extends JsonSerializable> ClusterManagementOperationStatusResult<V> checkStatus(
+      String opId) {
+    final CompletableFuture<V> status = executorManager.getStatus(opId);
+    if (status == null) {
+      throw new EntityNotFoundException("Operation id = " + opId + " not found");
+    }
+    ClusterManagementOperationStatusResult<V> result =
+        new ClusterManagementOperationStatusResult<>();
+    if (!status.isDone()) {
+      result.setStatus(ClusterManagementResult.StatusCode.IN_PROGRESS, "in progress");
+    } else {
+      try {
+        result.setResult(status.get());
+        result.setStatus(ClusterManagementResult.StatusCode.OK, "finished successfully");
+      } catch (InterruptedException | ExecutionException e) {
+        throw new RuntimeException(e);
```

Comment:
Just a quick comment here... you should probably reset the interrupt bit iif InterruptedException, before rethrowing it.

Reply:
InterruptedException can't actually be thrown by this `get()` since we only call it when `isDone()` is already true.  But, I will update the catch block anyway.